src/class-scripts.php

```diff
@@ -0,0 +1,201 @@
+	/**
+	 * Instance of Parsely class.
+	 *
+	 * @var Parsely
+	 */
+	private $parsely;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Parsely $parsely Instance of Parsely class.
+	 */
+	public function __construct( Parsely $parsely ) {
+		$this->parsely = $parsely;
+	}
+
+	/**
+	 * Register js scripts.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return void
+	 */
+	public function run(): void {
+		add_action( 'init', array( $this, 'register_js' ) );
+		add_action( 'wp_enqueue_scripts', array( $this, 'load_js_api' ) );
+		add_action( 'wp_enqueue_scripts', array( $this, 'load_js_tracker' ) );
+	}
+
+	/**
+	 * Register JavaScripts, if there's an API key value saved.
+	 *
+	 * @since 2.5.0
+	 *
+	 * @return void
+	 */
+	public function register_js(): void {
+		$parsely_options = $this->parsely->get_options();
+
+		if ( $this->parsely->api_key_is_missing() ) {
+			return;
+		}
+
+		wp_register_script(
+			'wp-parsely-tracker',
+			'https://cdn.parsely.com/keys/' . $parsely_options['apikey'] . '/p.js',
+			array(),
+			Parsely::get_asset_cache_buster(),
+			true
+		);
+
+		$api_script_asset = require plugin_dir_path( PARSELY_FILE ) . 'build/init-api.asset.php';
+		wp_register_script(
+			'wp-parsely-api',
+			plugin_dir_url( PARSELY_FILE ) . 'build/init-api.js',
+			$api_script_asset['dependencies'],
+			Parsely::get_asset_cache_buster(),
+			true
+		);
+
+		wp_localize_script(
+			'wp-parsely-api',
+			'wpParsely', // This globally-scoped object holds variables used to interact with the API.
+			array(
+				'apikey' => $parsely_options['apikey'],
+			)
+		);
+	}
+
+	/**
+	 * Enqueues the JavaScript code required to send off beacon requests.
+	 *
+	 * @since 2.5.0 Rename from insert_parsely_javascript
+	 *
+	 * @return void
+	 */
+	public function load_js_tracker(): void {
+		$parsely_options = $this->parsely->get_options();
+		if ( $this->parsely->api_key_is_missing() || $parsely_options['disable_javascript'] ) {
+			return;
+		}
+
+		global $post;
+		$display = true;
+		if ( in_array( get_post_type(), $parsely_options['track_post_types'], true ) && ! Parsely::post_has_trackable_status( $post ) ) {
+			$display = false;
+		}
+		if ( ! $parsely_options['track_authenticated_users'] && $this->parsely->parsely_is_user_logged_in() ) {
+			$display = false;
+		}
+		if ( ! in_array( get_post_type(), $parsely_options['track_post_types'], true ) && ! in_array( get_post_type(), $parsely_options['track_page_types'], true ) ) {
+			$display = false;
+		}
+
+		/**
+		 * Filters whether to enqueue the Parsely JavaScript tracking script from the CDN.
+		 *
+		 * If true, the script is enqueued.
+		 *
+		 * @since 2.5.0
+		 *
+		 * @param bool $display True if the JavaScript file should be included. False if not.
+		 */
+		if ( ! apply_filters( 'wp_parsely_load_js_tracker', $display ) ) {
+			return;
+		}
+
+		if ( ! has_filter( 'script_loader_tag', array( $this, 'script_loader_tag' ) ) ) {
+			add_filter( 'script_loader_tag', array( $this, 'script_loader_tag' ), 10, 3 );
+		}
```

Comment:
I have noticed that this construction is used in at least two places… Does it make sense to extract it into a separate helper method?

Reply:
Similar to the reply to @GaryJones above, I will do this on a separate PR, but it does make sense.